src/Domain/Insights/SyntaxCheck.php

```diff
@@ -36,38 +38,20 @@ public function getDetails(): array
 
     public function process(): void
     {
-        $phpPath = (string) Config::getExecutablePath('php');
-        $toExclude = array_map(
-            static fn (string $file): string => '--exclude ' . escapeshellarg($file),
-            array_merge($this->excludedFiles, LocalFilesRepository::DEFAULT_EXCLUDE)
-        );
-
-        $binary = sprintf(
-            '%s %s',
-            escapeshellcmd($phpPath),
-            escapeshellarg($this->vendorParentPathFind() . '/vendor/bin/parallel-lint')
-        );
-        if (DIRECTORY_SEPARATOR === '\\') {
-            $binary = $this->vendorParentPathFind() . '\vendor\bin\parallel-lint.bat';
-        }
-
-        $toAnalyse = '.';
-        if (\count($this->collector->getFiles()) === 1) {
-            // We ask to analyse one file, just proceed it
-            $files = $this->collector->getFiles();
-            $toAnalyse = \array_pop($files);
-        } elseif (rtrim($this->collector->getCommonPath(), DIRECTORY_SEPARATOR) !== getcwd()) {
-            $toAnalyse = $this->collector->getCommonPath();
-        }
+        $toAnalyse = $this->getTarget();
 
         $cmdLine = sprintf(
             '%s --no-colors --no-progress --json %s %s',
-            $binary,
-            implode(' ', $toExclude),
+            $this->getBinary(),
+            implode(' ', $this->getShellExcludeArgs()),
             $toAnalyse
         );
 
         $process = Process::fromShellCommandline($cmdLine);
+
+        if ($toAnalyse === '.' && getcwd() !== rtrim($this->collector->getCommonPath(), DIRECTORY_SEPARATOR)) {
+            $process->setWorkingDirectory($this->collector->getCommonPath());
```

Comment:
parallel-lint seems picky with paths. As far as I can tell, if the "target" is absolute it also expects the `--exclude` args to be absolute. When the target is relative, absolute paths don't work in `--exclude`.

By setting the working dir to the common path when it differs from $CWD, we avoid the need to pass the full path as the target.